src/data/transforms/shuffle.rs

```diff
@@ -74,6 +75,12 @@ impl Default for Shuffler<ThreadRng> {
 ///
 /// Under the hood this uses a Fisher-Yates shuffle.
 impl<R: Rng, T> Transformer<Matrix<T>> for Shuffler<R> {
+
+    #[allow(unused_variables)]
+    fn fit(&mut self, inputs: &Matrix<T>) -> Result<(), Error> {
+        unimplemented!();
```

Comment:
This is a tricky one.

I think we would prefer to just have the function be a no-op. My only concern would be that users might expect to be able to retrieve information about the transformation after `fit`ting - for example a list of row-indices to be swapped.

Otherwise without any additional documentation informing them of the panic I think almost all users would be confused and tripped up by this.